pkg/api/norman/customization/gke/listers.go

```diff
@@ -0,0 +1,197 @@
+package gke
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/rancher/norman/httperror"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/container/v1"
+	"google.golang.org/api/iam/v1"
+)
+
+func getOAuthClient(ctx context.Context, credentialContent string) (*http.Client, error) {
+	ts, err := google.CredentialsFromJSON(ctx, []byte(credentialContent), container.CloudPlatformScope)
+	if err != nil {
+		return nil, err
+	}
+
+	return oauth2.NewClient(ctx, ts.TokenSource), nil
+}
+
+func getComputeServiceClient(ctx context.Context, credentialContent string) (*compute.Service, error) {
+	client, err := getOAuthClient(ctx, credentialContent)
+
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := compute.New(client)
+
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+func getIamServiceClient(ctx context.Context, credentialContent string) (*iam.Service, error) {
+	client, err := getOAuthClient(ctx, credentialContent)
+
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := iam.New(client)
+
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+func getContainerServiceClient(ctx context.Context, credentialContent string) (*container.Service, error) {
+	client, err := getOAuthClient(ctx, credentialContent)
+
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := container.New(client)
+
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+func listMachineTypes(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
+	if cap.Zone == "" {
+		return nil, httperror.ServerError.Status, fmt.Errorf("zone is required")
```

Comment:
is `ServerError` the right error to use here? This would be caused by a malformed request so it would normally be a 4xx error.

Reply:
Ok, you are right, I propose `422` which is `httperror.InvalidOption`

https://pkg.go.dev/github.com/rancher/norman/httperror@v0.0.0-20210225010917-c7fd1e24145b#InvalidAction

Reply:
These need to be whatever they were in v1.